File: algorithms/0069.sqrtx/sqrtx.rs
impl Solution {
    pub fn my_sqrt(n: i32) -> i32 {
        let n = n as i64;
        let mut x = n.clone();
        while x * x > n {
            x = (x + n / x) / 2;
        }
        x as i32
    }
}

File: algorithms/0069.sqrtx/sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        if (x == 0) {
            return 0;
        }
        double x0 = x, C = x;
        while (true) {
            double xi = 0.5 * (x0 + C/x0);
            if (fabs(xi - x0) < 1e-6) {
                break;
            }
            x0 = xi;
        }
        return int(x0);
    }
};